lib/broadway_cloud_pub_sub/google_api_client.ex

```diff
@@ -71,8 +81,13 @@ defmodule BroadwayCloudPubSub.GoogleApiClient do
       adapter = Keyword.get(opts, :__internal_tesla_adapter__, Hackney)
       connection_pool = Keyword.get(opts, :__connection_pool__, :default)
 
+      retry =
+        [should_retry: &should_retry?/1]
+        |> Keyword.merge(Keyword.get(opts, :retry, @retry_opts))
```

Comment:
I think this will be more predictable behaviour, if you just want to overwrite say :delay, you'd still get our default max_retries. WDYT?

```suggestion
        [should_retry: &should_retry?/1]
        |> Keyword.merge(@retry_opts)
        |> Keyword.merge(Keyword.get(opts, :retry, []))
```